sys/mips/mips/pm_machdep.c

```diff
@@ -295,18 +313,30 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 		/* NOTREACHED */
 	}
 
-#ifdef CPU_CHERI
+#if __has_feature(capabilities)
 	/*
 	 * Install CHERI signal-delivery register state for handler to run
 	 * in.  As we don't install this in the CHERI frame on the user stack,
-	 * it will be (genrally) be removed automatically on sigreturn().
+	 * it will be (generally) be removed automatically on sigreturn().
 	 */
-	hybridabi_sendsig(td);
-#endif
+	regs->pc = (trapf_pc_t)catcher;
+	regs->c12 = regs->pc;
+	regs->csp = sfp;
+	regs->c17 = td->td_pcb->pcb_cherisignal.csig_sigcode;
 
+	/*
+	 * For now only change IDC if we were sandboxed. This makes cap-table
+	 * binaries work as expected (since they need cgp to remain the same).
+	 *
+	 * TODO: remove csigp->csig_idc
+	 */
+	if (cheri_is_sandboxed) {
+		regs->ddc = td->td_pcb->pcb_cherisignal.csig_ddc;
+		regs->idc = td->td_pcb->pcb_cherisignal.csig_idc;
+	}
+#else
 	regs->pc = (trapf_pc_t)catcher;
-	// FIXME: should this be an offset relative to PCC or an address?
-	regs->t9 = (register_t)(__cheri_offset intptr_t)regs->pc;
+	regs->t9 = (register_t)(intptr_t)catcher;
```

Comment:
For hybrid this should the offset of pcc/catcher.

Reply:
This is the no-CHERI-at-all-case, not hybrid.  hybrid is now in freebsd64_sendsig().